modules/angular-meteor-collection.js

```diff
@@ -53,14 +54,61 @@ angularMeteorCollection.factory('AngularMeteorCollection', [
       return allPromise;
     };
 
-    AngularMeteorCollection.remove = function (keys) {
-      // remove collection
-      if (!keys) _.pluck(this, '_id');
-      // remove single doc
-      keys = [].concat(keys);
+    AngularMeteorCollection._upsertDoc = function(doc, useUnsetModifier) { 
```

Comment:
I agree, it does look more "clean". And also I found this https://github.com/barbatus/angular-meteor/blob/master/modules/angular-meteor-object.js#L139. `collection` is not defined, it would be nice if you could fix this in your pull request. As I said I'm currently checking my changes, but I also think it would be a good idea to add tests to `$meteorObject`.